### PR TITLE
Add first pass of e2e vendor tests

### PR DESCRIPTION
**Pull request by imalao**
Add 6 vendor test scenarios:

1. signup.magic-link.public.spec.ts: Tests magic link claiming flow, various edge cases -- these are mostly skipped for now
2. settings.vendor.protected.spec.ts: Tests vendor account settings and security
3. profile-edit.vendor.protected.spec.ts: Tests vendor profile editing  -- these are mostly skipped for now
4. dashboard-checklist.vendor.protected.spec.ts: Tests dashboard completion tracking 
5. forgot-password.public.spec.ts: Tests password reset flow
6. profile-features.vendor.protected.spec.ts: Tests additional profile features

### CODE REVIEW

File: frontend/e2e/fixtures/supabase.setup.ts [hunk 1]
```diff
@@ -8,13 +8,27 @@
...
+const SESSION_DIR = path.join(import.meta.dirname, '.auth');
 
 setup('reset local supabase db', async () => {
   if (process.env.CI === 'true') {
     console.log('CI detected — skipping db reset (handled by workflow)');
     return;
   }
   execSync('npx supabase db reset', { stdio: 'inherit' });
+
+  // Clear cached session files — DB reset invalidates all refresh tokens,
```
Comment (imalao) [inline frontend/e2e/fixtures/supabase.setup.ts:23]: from claude: 

After supabase db reset, old session files are stale but the fixture reuses them. The vendor-protected project previously had NO matching test (pointed at a non-existent file), so this was never caught. I need to clear session files after DB reset.

File: frontend/e2e/vendor/dashboard-checklist.vendor.protected.spec.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    await expect(page.getByText('Client Photo')).toBeVisible();
+  });
+
+  test('dashboard shows Edit Your Profile card with title link', async ({ page }) => {
+    const editProfileLink = page.getByRole('link', { name: 'Edit Your Profile' });
+    await expect(editProfileLink).toBeVisible();
+    await expect(editProfileLink).toHaveAttribute('href', '/partner/dashboard/profile');
+  });
+
+  test('dashboard shows Badge Toolkit card', async ({ page }) => {
```
Comment (klee97) [inline frontend/e2e/vendor/dashboard-checklist.vendor.protected.spec.ts:51]: let's add the inquiries and performance cards too

**Review by klee97 — COMMENTED** [review]
left a comment, otherwise lgtm

File: frontend/e2e/auth/login.public.spec.ts [hunk 3]
```diff
@@ -13,20 +13,20 @@ test.describe('Login — guest', () => {
   });
 
   test('desktop navbar shows Login button when not logged in', async ({ page, isMobile }) => {
-    test.skip(isMobile, DESKTOP_ONLY_DESCRIPTION);
+    test.fixme(isMobile, DESKTOP_ONLY_DESCRIPTION);
```
Comment (klee97) [inline frontend/e2e/auth/login.public.spec.ts]: all the skips in this file are intentional!

File: frontend/e2e/favorites/favorites.public.spec.ts [hunk 4]
```diff
@@ -61,15 +61,15 @@ test.describe('Favorites — guest', () => {
   });
 
   test('mobile: clicking heart on vendor card shows login prompt', async ({ page, isMobile }) => {
-    test.skip(!isMobile, MOBILE_ONLY_DESCRIPTION);
+    test.fixme(!isMobile, MOBILE_ONLY_DESCRIPTION);
```
Comment (klee97) [inline frontend/e2e/favorites/favorites.public.spec.ts]: skips in this file are intentional

File: frontend/e2e/auth/login.public.spec.ts (hunk 3 above)
Comment (imalao) [reply frontend/e2e/auth/login.public.spec.ts]: ahhh thanks!! i totally mindlessly replaced skip to fixme

File: frontend/e2e/vendor/dashboard-checklist.vendor.protected.spec.ts (hunk 2 above)
Comment (imalao) [reply frontend/e2e/vendor/dashboard-checklist.vendor.protected.spec.ts:51]: Good call! Added!

**Review by klee97 — COMMENTED** [review]
LGTM!
